core/src/main/java/nl/nn/adapterframework/doc/model/ElementType.java

```diff
@@ -1,24 +1,149 @@
+			this.simpleName = clazz.getSimpleName();
+			if(clazz.isInterface()) {
+				for(Class<?> superInterface: clazz.getInterfaces()) {
+					InterfaceHierarchyItem superInterfaceHierarchyItem = new InterfaceHierarchyItem(superInterface);
+					parentInterfaces.put(superInterfaceHierarchyItem.getFullName(), superInterfaceHierarchyItem);
+				}
+			}
+		}
+
+		List<ElementType> findMatchingElementTypes(FrankDocModel model) {
+			ElementType currentMatch = model.findElementType(fullName);
+			if(currentMatch != null) {
+				return Arrays.asList(currentMatch);
+			}
+			List<ElementType> result = new ArrayList<>();
+			for(String parentKey: parentInterfaces.keySet()) {
+				result.addAll(parentInterfaces.get(parentKey).findMatchingElementTypes(model));
+			}
+			return result;
+		}
+	}
+
+	private final InterfaceHierarchyItem interfaceHierarchy;
+	private @Getter ElementType highestCommonInterface;
 
 	ElementType(Class<?> clazz) {
-		fullName = clazz.getName();
-		simpleName = clazz.getSimpleName();
+		interfaceHierarchy = new InterfaceHierarchyItem(clazz);
 		members = new HashMap<>();
 		this.fromJavaInterface = clazz.isInterface();
 	}
 
+	public String getFullName() {
+		return interfaceHierarchy.getFullName();
+	}
+
+	public String getSimpleName() {
+		return interfaceHierarchy.getSimpleName();
+	}
+
 	void addMember(FrankElement member) {
 		members.put(member.getFullName(), member);
 	}
+
+	FrankElement getSingletonElement() throws ReflectiveOperationException {
+		if(members.size() != 1) {
+			throw new ReflectiveOperationException(String.format("Expected that ElementType [%s] contains exactly one element", getFullName()));
+		}
+		return members.values().iterator().next();
+	}
+
+	void registerElementRole(ElementRole elementRole) {
+		elementRoles.add(elementRole);
+	}
+
+	void calculateHighestCommonInterface(FrankDocModel model) {
+		highestCommonInterface = this;
+		ElementType nextCandidate = highestCommonInterface.getNextCommonInterface(model);
+		while(nextCandidate != null) {
+			highestCommonInterface = nextCandidate;
+			nextCandidate = highestCommonInterface.getNextCommonInterface(model);
+		}
+		if(log.isTraceEnabled()) {
+			log.trace(String.format("ElementType [%s] has highest common interface [%s]", this.getFullName(), highestCommonInterface.getFullName()));
+		}
+	}
+
+	private ElementType getNextCommonInterface(FrankDocModel model) {
+		if(! fromJavaInterface) {
+			return null;
+		}
+		List<ElementType> candidates = new ArrayList<>();
+		for(String key: interfaceHierarchy.getParentInterfaces().keySet()) {
+			candidates.addAll(interfaceHierarchy.getParentInterfaces().get(key).findMatchingElementTypes(model));
+		}
+		if(candidates.isEmpty()) {
+			return null;
+		} else {
+			ElementType result = candidates.get(0);
+			if(candidates.size() >= 2) {
+				log.warn(String.format("There are multiple candidates for the next common interface of ElementType [%s], which are [%s]. Chose [%s]",
+						getFullName(),
```

Comment:
Geeft `getFullName()` altijd de juiste waarde terug voor `candidates.get(0)` ?
Moet dit niet `result.getFullName()` zijn?

Reply:
Hier wordt een foutmelding in de log geschreven. In die foutmelding moet staan op welk ElementType object de getNextCommonInterface() method wordt toegepast. Dat ElementType wordt hier gerefereerd.

Reply:
Ah, ik vroeg het mij af omdat in de log regel `Chose` staat... Misschien dat het duidelijke is/wordt als je er `elementType [...] fullName [...]` oid van maakt.